pkg/component/gardener/dashboard/configmap.go

```diff
@@ -127,6 +128,30 @@ func (g *gardenerDashboard) configMap(ctx context.Context) (*corev1.ConfigMap, e
 			},
 		}
 
+		if g.values.OIDC.CertificateAuthoritySecretRef != nil {
+			caSecret := &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      g.values.OIDC.CertificateAuthoritySecretRef.Name,
+					Namespace: g.namespace,
+				},
+			}
+
+			if err := g.client.Get(ctx, client.ObjectKeyFromObject(caSecret), caSecret); err != nil {
+				return nil, fmt.Errorf("failed to get referenced ca secret: %w", err)
+			}
+
+			caData, ok := caSecret.Data["ca.crt"]
+			if !ok {
+				return nil, fmt.Errorf("failed reading ca secret: missing ca.crt key")
+			}
+
+			if _, err := utils.DecodeCertificate(caData); err != nil {
+				return nil, fmt.Errorf("failed decoding ca certificate: %w", err)
```

Comment:
wording can be refined to make it clear that CA is invalid.
```suggestion
				return nil, fmt.Errorf("invalid ca certificate: failed decoding ca certificate: %w", err)
```